library/src/tensile_host.cpp

```diff
@@ -178,63 +172,70 @@ namespace
         Tensile::TensorDescriptor d{
             Tensile_To, {prob.m, prob.n, prob.batch_count}, {1, prob.ld_d, prob.stride_d}};
 
-        // Return the ContractionProblem
+        // The ContractionProblem
         Tensile::ContractionProblem tensileProblem{a,
                                                    aops,
                                                    b,
                                                    bops,
                                                    c,
-                                                   {},
+                                                   cops,
                                                    d,
-                                                   {},
+                                                   dops,
                                                    freeIndex,
                                                    batchIndex,
                                                    boundIndex,
                                                    value_category(*prob.beta)};
 
+        // If HPA is active, mark it as true
         if(sizeof(Tc) > sizeof(To))
             tensileProblem.setHighPrecisionAccumulate(true);
 
         return tensileProblem;
     }
 
-    // Tensile does not support float alpha and beta for HPA half
-    // We must convert alpha and beta from float to half
-    template <typename Ti, typename To, typename Tc>
-    struct HPA_workaround
+    /*************************************************************************
+     * Class for converting alpha and beta between rocBLAS and Tensile types *
+     * By default, alpha and beta are the same type as Tc compute_type       *
+     *************************************************************************/
+    template <typename Ti, typename To = Ti, typename Tc = To>
+    struct AlphaBeta
     {
-        using type = typename rocblas_to_tensile_type<Tc>::type;
-        static void copy(type* dst, const Tc* src)
+        using tensile_type = typename rocblas_to_tensile_type<Tc>::tensile_type;
+        static void copy(tensile_type* dst, const Tc* src)
         {
             static_assert(sizeof(*src) == sizeof(*dst),
```

Comment:
Using the types instead of *pointers it would be clearer.

Reply:
> Using the types instead of *pointers it would be clearer.

I disagree, because if you use types, you have to name the type in two places, while if you use the pointers, you're always sure you're using the type that it was declared as.

[Reference](https://wiki.sei.cmu.edu/confluence/display/c/EXP09-C.+Use+sizeof+to+determine+the+size+of+a+type+or+variable?focusedCommentId=88016344#comment-88016344)